unstructured_inference/inference/layout.py

```diff
@@ -412,20 +450,26 @@ def load_pdf(
                 layout.append(text_region)
         layouts.append(layout)
 
-    # Convert a PDF in small chunks of pages at a time (e.g. 1-10, 11-20... and so on)
-    info = pdf2image.pdfinfo_from_path(filename)
-    total_pages = info["Pages"]
-    images = []
-
-    for start_page in range(1, total_pages + 1, chunk_size):
-        end_page = min(start_page + chunk_size - 1, total_pages)
-        with tempfile.TemporaryDirectory() as tmpdir:
-            chunk_images = pdf2image.convert_from_path(
-                filename,
-                dpi=dpi,
-                first_page=start_page,
-                last_page=end_page,
-                output_folder=tmpdir,
-            )
-            images += chunk_images
```

Comment:
Is this the part of the code that was suspected to cause "too many files open" errors? If the `with` context manager automatically closes the temp directory after each iteration, I don't understand how this would result in the error. Let me know if I'm missing something.

Reply:
@six5532one Yes, this part of the code causes the issue. The `with` context manager automatically closes the temp directory after each iteration, but the chunk image files added to the 'images' variable remain open even after deletion.

Reply:
Thanks for explaining.